src/gap-skipper.js

```diff
@@ -0,0 +1,88 @@
+import Ranges from './ranges';
+
+let adaptiveSeeking;
+let gapSkipper;
+let seekingObject;
+
+gapSkipper = function(options) {
+  this.ready(() => {
+    seekingObject = { player: this,
+                      tech: this.tech_,
+                      seeking: false,
+                      consecutiveUpdates: 0,
+                      timer: null,
+                      playerState: null,
+                      lastRecordedTime: null,
+                      adaptiveSeeking: null };
+
+    // Allows us to mimic a waiting event in chrome
+    seekingObject.player.on('timeupdate', function() {
+      if (seekingObject.player.paused()) {
+        return;
+      }
+      let currentTime = seekingObject.player.currentTime();
+
+      if (seekingObject.consecutiveUpdates === 5 &&
+          currentTime === seekingObject.lastRecordedTime) {
+        // trigger waiting
+        if (seekingObject.playerState !== 'waiting') {
+          seekingObject.consecutiveUpdates = 0;
+          seekingObject.playerState = 'waiting';
+          seekingObject.tech.trigger('adaptive-seeking');
+        }
+      } else if (currentTime === seekingObject.lastRecordedTime) {
+        seekingObject.consecutiveUpdates++;
+      } else {
+        seekingObject.consecutiveUpdates = 0;
+        seekingObject.lastRecordedTime = currentTime;
+      }
+    });
+
+    // Don't listen for waiting while seeking
+    seekingObject.player.on('seeking', function() {
+      seekingObject.seeking = true;
+    });
+
+    // Listen for waiting when finished seeking
+    seekingObject.player.on('seeked', function() {
+      seekingObject.seeking = false;
+    });
+
+    seekingObject.player.on('playing', function() {
+      seekingObject.player.on('waiting', adaptiveSeeking);
+    });
+
+    seekingObject.tech.on('adaptive-seeking', adaptiveSeeking);
+
+    seekingObject.player.on('error', function() {
+      if (seekingObject.timer) {
+        clearTimeout(seekingObject.timer);
+      }
+    });
+  });
+};
+
+adaptiveSeeking = function() {
```

Comment:
you may just want to make this a method of seeking object so that you don't have to use a global reference